impeller/geometry/path_builder.cc

```diff
@@ -284,6 +284,9 @@ void PathBuilder::AddContourComponent(const Point& destination,
 }
 
 void PathBuilder::AddLinearComponent(const Point& p1, const Point& p2) {
+  if (ScalarNearlyEqual(p1.x, p2.x) && ScalarNearlyEqual(p1.y, p2.y)) {
```

Comment:
Do we know what the transform is? The skippable distance should take that into account for when we zoom in on a shape.

Reply:
The transform doesn't matter because we are hitting the limits of the float32 format.  So even if this is scaled up to be meaningful the points would be unstable.

Reply:
Right, we don't practically care about rendering correctly with a 1000000000x scale.

Reply:
.001 is nowhere near a 1000000000x scale. Scaling up by 100 can affect pixelization. Scaling by 1000 could leave a pixel gap or miss a pixel entirely.

Reply:
And "the limits of float32 format" - how are you defining that? It can handle 10^(+/-37) and it can represent pixels up to +/- 16 million before it loses sub-pixel accuracy.

If you want to take float precision into account then you should use the built-in ulp-based tests.

Note that large scales aren't necessariy for large zooms. A developer could create geometry in the 0.0->1.0 coordinate space and then scale it into place. On a 4k monitor that could involve scales larger than 1000x.

Reply:
> how are you defining that?

It's the standard definition for Impeller, `ScalarIsNearly` which uses the tolerance `kEhCloseEnough` which is 1e-3f.  If that isn't sufficient we should consider changing kEhCloseEnough (and ScalarIsNearly).